### PR TITLE
feature/inactive-series-group

### PR DESCRIPTION
**Pull request by pawelfus**
Inactive state: applied series.opacity for groups instead of path elements.

### CODE REVIEW

File: js/parts/Interaction.js [hunk 1]
```diff
@@ -1186,6 +1205,23 @@ extend(Series.prototype, /** @lends Highcharts.Series.prototype */ {
                         i = i + 1;
                     }
                 }
+
+                // For some types (pie, networkgraph, sankey) opacity is
+                // resolved on a point level
+                if (!series.options.inactiveOtherPoints) {
+                    [
+                        series.group,
+                        series.dataLabelsGroup
```
Comment (TorsteinHonsi) [inline js/parts/Interaction.js]: What about the `series.markerGroup`? Are we still looping over every `point.graphic` and setting its opacity? In that case, wouldn't it be more effective to set the series opacity on the marker group?

Comment (pawelfus) [reply js/parts/Interaction.js]: Yes, we still have `marker.states.inactive.something` option. Should we remove that?

Comment (TorsteinHonsi) [reply js/parts/Interaction.js]: We must make sure that when we set a whole _series_ to inactive (like a line series), we don't loop over the points and animate the opacity of each. In this case, setting the opacity on the group should be enough.

File: js/parts/Interaction.js [hunk 2]
```diff
@@ -1186,6 +1205,23 @@ extend(Series.prototype, /** @lends Highcharts.Series.prototype */ {
...
+                    ].forEach(
+                        function (group) {
+                            if (group) {
+                                group.attr({
```
Comment (TorsteinHonsi) [inline js/parts/Interaction.js]: Should be `.animate`. Both this problem and the above (individual handling of markers), can be seen in http://utils.highcharts.local/samples/#view/highcharts/demo/windbarb-series when hovering the wind barbs.

File: js/parts/Interaction.js (hunk 1 above)
Comment (pawelfus) [reply js/parts/Interaction.js]: Ok 👍

File: js/parts/Interaction.js (hunk 2 above)
Comment (pawelfus) [reply js/parts/Interaction.js]: Ok 👍

File: js/parts/Interaction.js (hunk 1 above)
Comment (pawelfus) [reply js/parts/Interaction.js]: Done

File: js/parts/Interaction.js (hunk 2 above)
Comment (pawelfus) [reply js/parts/Interaction.js]: Done

**Comment by TorsteinHonsi** [comment]
Works very well now, I've manually tested all the samples in the online demos. Found a couple minor issues, but I will merge this and handle them from the master.

**Pull request merged**